### ISSUE TITLE
Documentation: What case do the keys in `request.headers` use?

### ISSUE THREAD

**Issue by jakubholynet** [claudiajs/claudia-api-builder#15]
The API docs page mentions that `headers` is a dictionary of headers but does not describe the format of the keys. Are they just as the caller specified them, or are they standardised to a particular case (such as first capital, rest lower case)? Thanks!

**Comment by stojanovic** [comment]
They are not standardised in any particular way, just pass the headers the same way you would pass them in http call, ie.

```js
{
  headers: {
    'Content-type': 'application/json'
  }
}
```

**Comment by gojko** [comment]
I don't think API gateway specifies this, but from my limited investigation it looks as if they get down cased

**Comment by jakubholynet** [comment]
If we use standard [Node.js request than the headers are lower-cased](https://nodejs.org/api/http.html#http_message_headers).

If that is indeed the case, it would be nice to mention it in the documentation of `.headers` to simplify the life of future devs :-) 

It would also mean that https://github.com/claudiajs/claudia-api-builder/blob/master/docs/api.md#the-request-object has misleading code example:

```js
api.corsOrigin(function (request) {
    if (/claudiajs.com$/.test(request.headers.Origin)) {
        return request.headers.Origin; // <--- shouldn't this be lowercase "origin"?!
    }
    return '';
});```

**Comment by gojko** [comment]
given this is not properly defined, and we're working with a bunch of assumptions, the best thing to do would be to add another property (eg `.normalizedHeaders`), that would lowercase header names always, and use that if you care about normalization. The change would be something similar to https://github.com/claudiajs/claudia/blob/master/json-templates/apigw-params.txt#L43, and the test would be in https://github.com/claudiajs/claudia/blob/master/spec/rebuild-web-api-spec.js#L121 - if you want to submit a pull request with that, i'll merge it.

**Comment by gojko** [comment]
I've done some more testing, and the headers are definitely preserved (so my earlier comment on lowercasing is wrong). I've implemented a change that adds another field `.normalizedHeaders`, with header names lowercased. 

The field is available to APIs deployed with claudia 1.6.0 (so api builder version is not that important). Here are the release notes: https://claudiajs.com/news/2016/08/02/claudia-1.6.0.html

**Comment by jakubholynet** [comment]
Thank you! This is really useful when one needs to work with the headers.
